datarepo-clienttests/src/main/java/common/utils/AuthenticationUtils.java

```diff
@@ -90,6 +93,34 @@ public static GoogleCredentials getApplicationDefaultCredential() throws IOExcep
     return applicationDefaultCredential;
   }
 
+  public static GoogleCredentials getTestRunnerSACredentials() throws Exception {
+    String testRunnerSAFile = readTestRunnerSAEnvVariable();
+    if (testRunnerSAFile != null) {
+      ServiceAccountSpecification testRunnerServiceAccount =
+          ServiceAccountSpecification.fromJSONFile(testRunnerSAFile);
+      testRunnerServiceAccount.validate();
+
+      if (testRunnerSACredential != null) {
+        return testRunnerSACredential;
+      }
+
+      synchronized (lockTestRunnerSACredential) {
+        File jsonKey = testRunnerServiceAccount.jsonKeyFile;
+        testRunnerSACredential =
+            ServiceAccountCredentials.fromStream(new FileInputStream(jsonKey))
+                .createScoped(
+                    Collections.singletonList("https://www.googleapis.com/auth/cloud-platform"));
+      }
+      return testRunnerSACredential;
+    }
+    return getApplicationDefaultCredential();
```

Comment:
```suggestion
    // otherwise, return the default `${GOOGLE_APPLICATION_CREDENTIALS}`
    return getApplicationDefaultCredential();
```

Reply:
Added a detailed comment - let me know if it makes sense.